cirq-core/cirq/optimizers/two_qubit_gate_compilation.py

```diff
@@ -0,0 +1,484 @@
+# pylint: disable=wrong-or-nonexistent-copyright-notice
+"""Attempt to tabulate single qubit gates required to generate a target 2Q gate
+with a product A k A."""
+from functools import reduce
+from typing import Tuple, Sequence, List, NamedTuple
+
+from dataclasses import dataclass
+import numpy as np
+
+import cirq
+from cirq import value
+from cirq._compat import proper_repr, proper_eq
+from cirq.optimizers.two_qubit_gate_math_utils import (
+    kak_vector_infidelity,
+    vector_kron,
+    weyl_chamber_mesh,
+    random_qubit_unitary,
+    kak_vector_to_unitary,
+)
+
+_SingleQubitGatePair = Tuple[np.ndarray, np.ndarray]
+
+
+class TwoQubitGateCompilation(NamedTuple):
```

Comment:
`TwoQubitGateCompilation` is a very generic name. How about we rename this to `TwoQubitGateTabulationResult` ? I'm also open to suggestions here -- naming is hard!